virttest/qemu_devices/qcontainer.py

```diff
@@ -1018,6 +1023,27 @@ def machine_q35(cmd=False):
                 devices.append(qdevices.QStringDevice('fdc',
                                                       child_bus=qdevices.QOldFloppyBus('floppy'))
                                )
+
+            # add default pcie root port plugging pcie device
+            port_name = '%s-0' % root_port_type
+            port_params = {
+                'type': root_port_type,
+                # reserve slot 0x0 for plugging in  pci bridge
+                'reserved_slots': '0x0'}
+            root_port = self.pcic_by_params(port_name, port_params)
+            if root_port_type == 'pcie-root-port':
+                root_port.set_param('multifunction', 'on')
+                root_port.set_param('port', hex(1))
```

Comment:
Looks like an extra bus is needed to help allocate the port number, what do you think?

Reply:
let's do it in feature, team wait for this fix for test.

Reply:
It's first root root port for Q35 machine, other root port insert before add device in QPCIEBus, and latest_port_index is 1 in  QPCIEBus，new port index is __latest_port_index +1.